api/src/main/java/org/apache/iceberg/expressions/Binder.java

```diff
@@ -91,12 +91,21 @@ static Expression bind(StructType struct,
   }
 
   public static Set<Integer> boundReferences(StructType struct, List<Expression> exprs, boolean caseSensitive) {
+    return boundReferences(struct, exprs, caseSensitive, false);
+  }
+
+  public static Set<Integer> boundReferences(
+      StructType struct, List<Expression> exprs, boolean caseSensitive, boolean alreadyBound) {
```

Comment:
I don't think it is a good idea to add a second boolean argument to this method. It is confusing enough with just one.

How about using a different method name for this and then renaming this to a be a private internal implementation?

Reply:
I changed this method name to `exprReferences` (please let me know if you have a better name), but I still need to keep this as `public` because I need to access this method from `ParquetBloomRowGroupFilter`, which is in a different package.

Reply:
Maybe just `references` since it doesn't bind?

Reply:
Sounds good. Will change.